__tests__/components/controls/ButtonCopy.spec.js

```diff
@@ -0,0 +1,60 @@
+import ButtonCopy from '../../../src/components/controls/ButtonCopy.vue';
+import UI from '../../../src/store/ui';
+import { createLocalVue, shallowMount } from '@vue/test-utils';
+import Vuex from 'vuex';
+
+const setupStoreMount = () => {
+	const uiModule = {
+		namespaced: true,
+		getters: {
+			getNameByKey: UI.getters.getNameByKey
+		}
+	};
+
+	const store = new Vuex.Store({
+		modules: {
+			ui:uiModule
+		}
+	});
+
+	const propsData = {
+		value: 'copy text',
+		successMessage: 'success'
+	};
+
+	return shallowMount(ButtonCopy, {
+		store,
+		localVue,
+		propsData,
+		mocks: {
+			$bvToast: {
+				toast: jest.fn()
+			}
+		}
+	});
+};
+
+const localVue = createLocalVue();
+localVue.use(Vuex);
+
+describe('Create ButtonCopy component', () => {
+	// Arrange
+	const wrapper = setupStoreMount();
+
+	it('processes props data', ()=> {
+		// Assert:
+		expect(wrapper.vm.value).toMatch('copy text');
+		expect(wrapper.vm.successMessage).toMatch('success');
+	});
+
+	it('click on copy button', () =>{
+		// Arrange
+		document.execCommand = jest.fn();
+
+		// Act:
+		wrapper.vm.onCopyClick();
+
+		// Assert:
+		expect(document.execCommand).toHaveBeenCalledWith('copy');
```

Comment:
try to test toast messages. also try to add test for error case.

Reply:
Yeah, I did try that, but it not working.
I will try again 👍🏼

Reply:
I added unit test individual for toast `errorMsg` and `successMsg` method.

try to work on this, but no luck, not sure I'm in the right direction here.
I mocked `copyTextToClipboard` to assert the function is called, but not working.

still look around how to solve this.

```js
onCopyClick () {
	helper.copyTextToClipboard(this.value) // Mock this
		.then(() => this.successMsg()) // fail assert function is called
		.catch(() => this.errorMsg()); // fail assert function is called
},
```

Reply:
I found the problem now, all good :)